lib/plugins/aws/package/compile/functions.js

```diff
@@ -650,38 +648,60 @@ class AwsCompileFunctions {
     cfResources[this.provider.naming.getLambdaEventConfigLogicalId(functionName)] = resource;
   }
 
+  getDestinationsArn(destinationsProperty) {
+    if (typeof destinationsProperty === 'object') {
+      return destinationsProperty.arn;
+    }
+    return destinationsProperty.startsWith('arn:')
+      ? destinationsProperty
+      : this.provider.resolveFunctionArn(destinationsProperty);
+  }
+
   // Memoized in a constructor
   ensureTargetExecutionPermission(functionAddress) {
```

Comment:
Let's maybe replace the name `functionAddress` with something more generic? It's a bit confusing at the moment as it's not really a function address in most cases

Reply:
important.
sorry for my lack of consideration.😣